torch/fx/graph.py

```diff
@@ -848,6 +848,59 @@ def _tensor_annotation(t: torch.Tensor) -> str:
 # 2. In the FX graph, we need to access 2 attributes - in_spec and out_spec.
 #    Since we can't access .graph within the FX forward, we need to copy the attribute to the module.
 # 3. We currently can't register the pytree imports with `add_global` - not sure why.
+class _BoxedCodeGen(CodeGen):
+    """
+    CodeGen subclass that generates code using the "boxed" calling convention.
+
+    The boxed calling convention takes a single list argument and clears it
+    after extracting the arguments, which allows for early deallocation of
+    input tensors.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def process_inputs(self, args_list: Any) -> Any:
+        """
+        Process boxed inputs. Expects a list of arguments.
+        """
+        return args_list
+
+    def process_outputs(self, out: Any) -> Any:
+        """
+        Process outputs - just return as-is for boxed calling convention.
+        """
+        return out
+
+    def gen_fn_def(
+        self, free_vars, maybe_return_annotation, *, expanded_def: bool = False
+    ):
+        """
+        Generate function definition for boxed calling convention.
+
+        Instead of taking individual arguments, the generated function takes
+        a single 'args_list' parameter, extracts placeholder values from it,
+        and clears the list.
+        """
+        # Generate the function signature with args_list parameter
+        fn_def = f"def {self._func_name}(self, args_list){maybe_return_annotation}:"
+
+        if len(free_vars) > 0:
```

Comment:
empty containers are falsey, non-empty containers are truthy

(why don't people ever rely on that anymore?)